certbot-nginx/certbot_nginx/configurator.py

```diff
@@ -889,14 +1043,39 @@ def _test_block_from_block(block):
     parser.comment_directive(test_block, 0)
     return test_block[:-1]
 
+
+def _wildcard_domain(domain):
+        """
+        Checks if domain is a wildcard domain
+
+        :param str domain: Domain to check
+
+        :returns: If the domain is wildcard domain
+        :rtype: bool
+        """
+        if isinstance(domain, six.text_type):
+            wildcard_marker = u"*."
+        else:
+            wildcard_marker = b"*."
+        return domain.startswith(wildcard_marker)
+
+
 def _redirect_block_for_domain(domain):
+    updated_domain = domain
+    match_symbol = '='
+    if _wildcard_domain(domain):
+        match_symbol = '~'
+        updated_domain = updated_domain.replace('.', '\.')
+        updated_domain = updated_domain.replace('*', '.+')
```

Comment:
This needs to be restricted a bit more. For example, if `domain` is `*.example.org`, this regex will match `a.b.example.org` which isn't covered by a cert for the given domain. I think using `[^.]+` instead of `.+` here should work.

Reply:
I agree, done.